ConsoleApp1/Shop.cs

```diff
@@ -0,0 +1,317 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConsoleApp1
+{
+    internal class Shop
+    {
+        private  List<IProduct> products = new List<IProduct>();
+        private  List<IBuyer> buyers = new List<IBuyer>();
+        private  List<IReceipt> receipts = new List<IReceipt>();
+
+        /// <summary>
+        /// Add new product to the shop
+        /// </summary>
+        /// <param name="product"></param>
+        public void AddProduct(IProduct product)
+        {
+            products.Add(product);
+        }
+
+        /// <summary>
+        /// Add quantity of goods to the existed product in the shop
+        /// </summary>
+        /// <param name="product"></param>
+        /// <param name="quantity"></param>
+        public void AddProductQuantity(IProduct product, int quantity)
+        {
+            int index;
+            
+            if (ContainProduct(product, out index))
```

Comment:
you can declare index inside method ContainProduct` ContainProduct(product, out int index)`

Reply:
thanks, done